src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/RequestLoggingServiceBehavior.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// 
+
+using System;
+using System.Collections.ObjectModel;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.ServiceModel.Dispatcher;
+using System.Text;
+
+namespace WcfService
+{
+    public class RequestLoggingServiceBehavior : IServiceBehavior
+    {
+        /* This service behavior allows adding logging to a service at the earliest possible point. This
+           is done by using a custom ServiceAuthorizationManager. This behavior chains any existing authorization
+           manager to preserve the authorization mechanism in use. It can be used in an EndpointResource by overriding
+           the ModifyHost method. It is best to add the behavior after base.ModifyHost has been called otherwise
+           any other ServiceAuthorizationManager that gets installed could replace this.
+           Sample usage:
+               protected override void ModifyHost(ServiceHost serviceHost, ResourceRequestContext context)
+               {
+                   base.ModifyHost(serviceHost, context);
+                   var loggingBehavior = new RequestLoggingServiceBehavior(RequestLoggingServiceBehavior.SimpleConsoleLogger);
+                   serviceHost.Description.Behaviors.Add(loggingBehavior);
+               }
+
+           The constructor takes a delegate of type RequestLoggingServiceBehavior.LogRequest. There is a simple static logger that can
+           be used that's available as RequestLoggingServiceBehavior.SimpleConsoleLogger and is used in the example above. As the logger
+           is a delegate, you can add multiple loggers and they will be called one after the other. To add more loggers, use the AddLogger
+           method.
+           */
+        private LogRequest _logCallback;
+
+        public delegate void LogRequest(OperationContext operationContext, ref Message message, bool authenticated);
+        public RequestLoggingServiceBehavior() { }
```

Comment:
I know we're not pruning on this iteration, but if you're keeping track, I don't think this is being used.  We can keep it for now.